src/ucai/core/databricks.py

```diff
@@ -215,6 +216,150 @@ def create_function(
+                \"\"\"
+                Adds the length of a string to an integer.
+                
+                Args:
+                    a (int): The integer to add to.
+                    b (str): The string whose length will be added.
+
+                Returns:
+                    float: The sum of the integer and the string length.
+                \"\"\"
+                return a + len(b)
+            ```
+            - If docstrings do not conform to Google-style for specifying arguments descriptions, parameter descriptions
+             will default to `"Parameter <name>"`, and no further information will be provided in the function comment
+             for the given parameter.
+
+            For examples of Google docstring guidelines, see
+            [this link](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
+
+        3. **External Dependencies**:
+            - Unity Catalog UDFs are limited to Python standard libraries and Databricks-provided libraries. If your 
+            function relies on unsupported external dependencies, the created UC function may fail at runtime.
+            - It is strongly recommended to test the created function by executing it before integrating it into 
+            GenAI or other tools.
+
+        **Function Metadata**:
+        - Docstrings (if provided and Google-style) will automatically be included as detailed descriptions for 
+        function parameters as well as for the function itself, enhancing the discoverability of the utility of your
+        UC function.
+
+        **Example**:
+        ```python
+        def example_function(x: int, y: int) -> float:
+            \"\"\"
+            Multiplies an integer by the length of a string.
+            
+            Args:
+                x (int): The number to be multiplied.
+                y (int): A string whose length will be used for multiplication.
+            
+            Returns:
+                float: The product of the integer and the string length.
+            \"\"\"
+            return x * len(y)
+
+        client.create_python_function(
+            func=example_function,
+            catalog="my_catalog",
+            schema="my_schema"
+        )
+        ```
+
+        Args:
+            func (Callable): The Python function to convert into a UDF.
+            catalog (str): The catalog name in which to create the function.
+            schema (str): The schema name in which to create the function.
+
+        Returns:
+            FunctionInfo: Metadata about the created function, including its name and signature.
+        """
+
+        if not callable(func):
+            raise ValueError("The provided function is not callable.")
+
+        try:
+            sql_function_body = generate_sql_function_body(func, catalog, schema)
+        except Exception as e:
+            raise RuntimeError(f"Failed to generate SQL function body for {func.__name__}") from e
```

Comment:
do we need this try-catch?

Reply:
haha I was just doing more e2e testing and was thinking the same thing... it's just a wrapper around a more informative exception :) I'm going to remove it

Reply:
@harupy Updated to remove the exception and also added better error messages for the return types (they were too vague for the collections, Union, and Any types). Error messages are much more clear about what is wrong and provide concrete guidance now.